src/SFA.DAS.EmployerApprenticeshipsService.Application/Commands/UnsubscribeNotification/UnsubscribeNotificationHandler.cs

```diff
@@ -0,0 +1,87 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+using MediatR;
+
+using SFA.DAS.EAS.Application.Validation;
+using SFA.DAS.EAS.Domain.Data.Repositories;
+using SFA.DAS.EAS.Domain.Models.UserProfile;
+using SFA.DAS.NLog.Logger;
+using SFA.DAS.Notifications.Api.Client;
+using SFA.DAS.Notifications.Api.Types;
+
+namespace SFA.DAS.EAS.Application.Commands.UnsubscribeNotification
+{
+    public class UnsubscribeNotificationHandler : AsyncRequestHandler<UnsubscribeNotificationCommand>
+    {
+        private readonly IValidator<UnsubscribeNotificationCommand> _validator;
+        private readonly INotificationsApi _notificationsApi;
+        private readonly IUserRepository _userRepository;
+        private readonly IAccountRepository _accountRepository;
+        private readonly ILog _logger;
+
+        public UnsubscribeNotificationHandler(
+            IValidator<UnsubscribeNotificationCommand> validator,
+            INotificationsApi notificationsApi,
+            IUserRepository userRepository,
+            IAccountRepository accountRepository,
+            ILog logger)
+        {
+            _validator = validator;
+            _notificationsApi = notificationsApi;
+            _userRepository = userRepository;
+            _accountRepository = accountRepository;
+            _logger = logger;
+        }
+
+        protected override async Task HandleCore(UnsubscribeNotificationCommand command)
+        {
+            _validator.Validate(command);
+            
+            var settings = await _accountRepository.GetUserAccountSettings(command.UserRef);
+            var setting = settings.FirstOrDefault(m => m.AccountId == command.AccountId);
```

Comment:
Should there only ever be 1 setting per user per account? Should this be SingleOrDefault?